lib/adf_diag.py

```diff
@@ -533,30 +532,40 @@ def call_ncrcat(cmd):
 
                 #Notify user of new time series file:
                 print(f"\t - time series for {var}")
+                
+                #Variable list starts with just the variable
+                ncrcat_var_list = f"{var}"
 
                 #Determine "ncrcat" command to generate time series file:
+                if 'date' in hist_file_ds[var].dims:  
+                    ncrcat_var_list = ncrcat_var_list + ",date"
+                if 'datesec' in hist_file_ds[var].dims:  
+                    ncrcat_var_list = ncrcat_var_list + ",datesec"
+
                 if has_lev and vert_coord_type:
-                    if vert_coord_type == "hybrid":
-                        cmd = ["ncrcat", "-O", "-4", "-h", "-v",
-                               f"{var},hyam,hybm,hyai,hybi,PS"] + \
-                               hist_files + ["-o", ts_outfil_str]
-                    elif vert_coord_type == "height":
+                    #PS might be in a different history file. If so, continue without error.
+                    ncrcat_var_list = ncrcat_var_list + f",hyam,hybm,hyai,hybi"
+
+                    if 'PS' in hist_file_ds[var].dims:  
+                        ncrcat_var_list = ncrcat_var_list + ",PS"
+                    else:
+                        print(f"WARNING: PS not found in history file. It might be needed at some point.")
+                    #End if
+
+                    if vert_coord_type == "height":
                         #Adding PMID here works, but significantly increases
                         #the storage (disk usage) requirements of the ADF.
                         #This can be alleviated in the future by figuring out
                         #a way to determine all of the regridding targets at
                         #the start of the ADF run, and then regridding a single
                         #PMID file to each one of those targets separately. -JN
-                        cmd = ["ncrcat", "-O", "-4", "-h", "-v", f"{var},PMID,PS"] + \
-                                hist_files + ["-o", ts_outfil_str]
+                        ncrcat_var_list = ncrcat_var_list + ",PMID"
```

Comment:
Should we have a check here to see if `PMID` is actually in the history file, and if not print a warning, similar to what was done for `PS`?